### DISCUSSION TITLE
Time range: from parameter in the get_quote function

### DISCUSSION THREAD

**Discussion by andreltr** [serkor1/cryptoQuotes#18] [General]
Hello,

Is there a limit in the from & to values in the get_quote function?  I may have read that the from past time parameter would be limited to one year in the past only.  If true I'd suggest no limit for this parameter in order to enable analysis of long term positions.

Best.

**Comment by serkor1** [comment]
Hi @andreltr,

There is no limit to the `from`- and `to`-parameters imposed by {cryptoQuotes} per se. However, the default returned values are 200(ish) rows if either of these are `NULL`. Below is an example,

``` r
library(cryptoQuotes)
```

``` r
nrow(
  BTC <- get_quote(
    ticker = "BTCUSDT",
    from   = Sys.Date() - 500,
    source = "bybit"
  )
)
#> [1] 201
```

Here we extract data from from `Sys.Date() - 500`, as the `to`-parameter is `NULL` it only returns 200 rows of data. However, if you specify the `to`-parameter also, you will get 500 rows of data. See below,

``` r
nrow(
  BTC <- get_quote(
    ticker = "BTCUSDT",
    from   = Sys.Date() - 500,
    to     = Sys.Date(),
    source = "bybit"
  )
)
#> [1] 501
```

The only imposed limit in {cryptoQuotes} is on exchange level. You can see the documentation on the [wiki](https://github.com/serkor1/cryptoQuotes/wiki/Limits-and-Restrictions) about the limits imposed by each exchange.

The least restrictive API is Binance, followed by Bybit. But depending on your location these might not be available without a VPN.

<sup>Created on 2024-06-23 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

**Reply by andreltr** [reply]
Excellent! 
Thank you!
